src/mlir_build.cpp

```diff
@@ -138,11 +138,23 @@ boost::filesystem::path MiirBuildViaHip(boost::optional<TmpDir>& tmp_dir,
 void MiirGenLaunchParams(const std::string& params, size_t& local_size, size_t& global_size)
 {
     AutoMiirHandle handle(params);
-    miirLowerInit();
     auto status = miirLowerTuningParams(handle());
     check_miir_error(status, "miirLowerTuningParams");
     miirGetExecutionDims(handle(), &global_size, &local_size);
     check_miir_error(status, "miirGetExecutionDims");
 }
 
+void MiirGenBin(const std::string& params, std::vector<char>& buffer)
+{
+    AutoMiirHandle handle(params);
```

Comment:
Not related to this PR, but `AutoMiirHandle` can be copied which will lead to double frees. This really should just use `MIOPEN_MANAGE_PTR`, which will create a `std::unique_ptr`(which is movable and not copyable) to manage the opaque object. Perhaps we can should open issue to address this.

Reply:
@pfultz2 Thanks for observing this! I agree that making it like a `unique_ptr` to turn copy into compile time error is a more robust approach. Let me open the issue and leave another comment here.

Reply:
The programmer who copies guard objects should be killed.

Reply:
[Recommendation] If in doubt, just disable copying by
```C++
  AutoMiirHandle(const AutoMiirHandle&) = delete;
  void operator=(const AutoMiirHandle&) = delete;
```
in https://github.com/ROCmSoftwarePlatform/MIOpen/blob/8fe06e67a16b7261d59d70a0ae75d83e678ad788/src/mlir_build.cpp#L40-L48

Reply:
Addressed in 8fc3457